openff/toolkit/utils/exceptions.py

```diff
@@ -290,3 +290,9 @@ class DuplicateVirtualSiteTypeException(OpenFFToolkitException):
 
 class CallbackRegistrationError(OpenFFToolkitException, TypeError):
     """Error raised when callback registration fails."""
+
+
+class MissingIndexedAttributeError(
+    OpenFFToolkitException, IndexError, KeyError, AttributeError
+):
+    """Error raised when an indexed attribute does not exist"""
```

Comment:
Non-blocking: We should plan to remove the inheritance from `IndexError` and `KeyError` at some point but I haven't thought about how to communicate/implement this. I wouldn't want a message warning users to not catch as `IndexError`/`KeyError` if they're not doing that, or even close to that specific case. If it's not tractable it's probably okay to just leave the other errors in long-term.